Enhancement: follow system theme

just-max opened the pull request:
More or less like in #134 .

**All** of the code is still in the header because I wasn't sure how to get access to the Alpine.js object from the webpack.

Naming convention: "theme mode": whether the theme is set manually to light/dark or to automatic (that's what github calls it).

Visually I tried to follow the rest of the header's design language: hovering increases contrast (text-5 to text-1), the dropdown is basically the same as the one for notifications and the selected mode is marked with a darker background as on the semester selection widget.

Outstanding:
- is there a way to notify users? or should they just notice by themselves :)

Next steps?
- GitHub has a nice smooth transition between light and dark modes. Just setting `transition-colors duration-200` on the body doesn't really do it (not everything transitions smoothly, and it conflicts with the use of transitions for hovering).

resolves #134, resolves #133

just-max committed:
enhance theme selector with follow system theme option

just-max committed:
add theme selector for mobile, too

joschahenningsen commented:
Amazing, thanks a ton! This works as expected. 

> All of the code is still in the header

Imo absolutely fine to do it like this.

> is there a way to notify users?

I can create a notification for the tiny bell.

![image](https://user-images.githubusercontent.com/44805696/166866241-674106e6-a214-491e-8ce4-3b0a1998ef73.png)

It feels a little unintuitive to have a screen as a theme switcher button. Could we instead 
- show the current icon based on the system preference if system is selected (e.g. moon if my system prefers dark mode) **or**
- always show a generic icon like `fa-circle-half-stroke` if system is selected?

just-max committed:
switcher shows active theme when following system theme

just-max committed:
fix rounded corners in header dropdowns

on medium screens, the regular desktop header is shown, however the lg:rounded-* classes only add rounded corners on large screens and above

just-max commented:
:+1: to changing the switcher icon. I implemented suggestion 1:

![auto_dark](https://user-images.githubusercontent.com/17551908/167267682-d8760dd6-539a-41a5-a95f-882b67f233cd.gif)

joschahenningsen reviewed (approved):
Works and looks great, thanks a ton! :100:

just-max commented:
Thanks, no problem :)
